packages/adminrouter/extra/src/test-harness/modules/mocker/endpoints/mesos.py

```diff
@@ -294,6 +142,64 @@
 }
 
 
+def slave_from_template(sid, ip, port):
+    """Create a Mesos slave entry basing on the supplied data and the template
+
+    Arguments:
+        sid (string): slave ID that the new slave should have
+        port (string): TCP/IP port that the new slave should pretend to listen on
+        ip (string): IP address that the new slave hould pretend to listen on
+
+    Returns:
+        Slave dict mimicing the one returned by Marathon
+    """
+    res = copy.deepcopy(SLAVE_TEMPLATE)
+    res['id'] = sid
+    res['pid'] = "slave(1)@{0}:{1}".format(ip, port)
+    res['hostname'] = ip
+
+    return res
+
+SLAVE1_ID = "de1baf83-c36c-4d23-9cb0-f89f596cd6ab-S1"  # noqa: E305
+SLAVE2_ID = "de1baf83-c36c-4d23-9cb0-f89f596cd6ab-S0"  # noqa: E305
+SLAVE3_ID = "35f210bb-bb58-4559-9932-b62619e72b6d-S0"  # noqa: E305
+SLAVE_EXTRA_ID = "8ad5a85c-c14b-4cca-a089-b9dc006e7286-S2"  # noqa: E305
```

Comment:
> What is an extra slave? (maybe a comment would help)

I will add some more information to https://github.com/dcos/dcos/blob/master/packages/adminrouter/extra/src/test-harness/modules/mocker/endpoints/mesos.py#L341-L345 : `This is agent that can be enabled using 'enable_extra_slave' call. Normally it's not present in the mocked '/state-summary' output`

> How were these created? or... how do I add the ID of a fourth slave?

See above (https://github.com/dcos/dcos/blob/master/packages/adminrouter/extra/src/test-harness/modules/mocker/endpoints/mesos.py#L341-L345)

> Am I mixing systems when I think of the effort to change from "slave" to "agent" terminology?

No, you do not - we should be standardising on `agent`. I have freed all the slaves in the test harness.

> Perhaps this would be better as MESOS_SLAVE_1 etc.

This is already implied by the Mock name/module name.